Add support for serialize => false on group and save_default => false on field

pdufour opened the pull request:
Adding `serialize => false` on the field group means if you have a field named `foo` in a group `bar`, it gets saved under the meta value `foo` instead of `bar["foo"]`. The serialization behavior is still the default, though.

If `save_default` is `false` on a field, any time the value == the default value, that meta row does not get saved (or it gets deleted if it already exists).

pdufour committed:
Add support for serialize => false and save_default => false

Unserialized values are saved on the root, meaning if you have a field named "foo" in a group "bar",
it gets saved under the meta value "foo" instead of "bar["foo"]".

If save_default is false any time the value == the default value, that meta row does not get saved (or gets
deleted if it already exists).

R4c00n commented:
Thanks for your PR. Looks good, but could you please refactor your snake_case variable namings to camelCase, so that they fit to our code styling conventions and I can accept the PR?

pdufour committed:
Small bug fixes

- Change snake_case to camelCase
- Change save_defaults to saveDefaults
- Used trimmed version of strings to check for default equality
- Remove debug die
- Prevent deleting of meta row at any time when there is not a new $_POST value set

pdufour commented:
Thanks. I changed to camelCase and made a few other bug fixes.
